src/bcrypt_pbkdf.c

```diff
@@ -127,7 +127,10 @@ bcrypt_pbkdf(const char *pass, size_t passlen, const uint8_t *salt,
     memcpy(countsalt, salt, saltlen);
 
     /* collapse password */
-    (void)libssh2_sha512_init(&ctx);
+    if(!libssh2_sha512_init(&ctx)) {
+        free(countsalt);
+        return -1;
+    }
     libssh2_sha512_update(ctx, pass, passlen);
     libssh2_sha512_final(ctx, sha2pass);
```

Comment:
Perhaps this is a separate task, but should we not also change libssh2_sha*_update and libssh2_sha*_final to return int and check their return values?

Reply:
That would be useful, I agree. I'd make this a separate PR when concluding the pending ones just to keep fewer balls in the air.